### ISSUE TITLE
Metrics.json and TensorBoard only tracks every fourth evaluation

### ISSUE THREAD

**Issue by fromm1990** [facebookresearch/detectron2#1714]
Hi, first of all thanks for this very useful framework!

Currently I'm using the built in `COCOEvaluator`. The evaluator runs for every `EVAL_PERIOD` iterations, 1225 in this case. The `log.txt` file also contains information of every evaluation, first record at 1224 (starts at 0) next at 2449 etc. However, the metric.json file and TensorBoard only contains records for every fourth test, i.e. first at 4899 iteration, next at 9799 etc.

I'm not sure why the metric.json file and TensorBoard only shows data for every fourth test/evaluation?
 
## Instructions To Reproduce the Issue:

```python
class DarknetTrainer(DefaultTrainer):

    @classmethod
    def build_evaluator(cls, cfg, dataset_name, output_folder=None):
        if output_folder is None:
            output_folder = os.path.join(cfg.OUTPUT_DIR, 'inference')
        return COCOEvaluator(dataset_name, cfg, False, output_folder)

def setup(args) -> CfgNode:
    """
    Create configs and perform basic setups.
    """
    cfg = get_cfg()
    cfg.merge_from_file(args.config_file)
    cfg.merge_from_list(args.opts)

    # Create a subdirectroy for this run
    cfg.OUTPUT_DIR = os.path.join(cfg.OUTPUT_DIR, datetime.now().strftime("%Y-%m-%d-%H%M%S"))
    os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)

    cfg.freeze()
    
    """
    Perform some basic common setups at the beginning of a job, including:
        * Set up the detectron2 logger
        * Log basic information about environment, cmdline arguments, and config
        * Backup the config to the output directory
    """
    default_setup(cfg, args)
    return cfg

def main(args):
    cfg = setup(args)

    # Register all the datasets
    datasetManager= DarknetDatasetManager()
    for dataset in args.dataset:
        datasetManager.register(dataset[0], dataset[1], dataset[2], dataset[3])

    trainer = DarknetTrainer(cfg)

    trainer.resume_or_load(resume=False)
    trainer.train()

if __name__ == "__main__":
    parser = default_argument_parser()

    parser.add_argument(
        "--dataset",
        nargs=4,
        action="append",
        metavar=("img_dir_path", "label_dir_path", "dataset_name", "names_file_path"),
        help="Adds a dataset to the list of registered datasets.",
        required=True)

    args = parser.parse_args()

    print("Command Line Args:", args)

    launch(
        main,
        args.num_gpus,
        num_machines=args.num_machines,
        machine_rank=args.machine_rank,
        dist_url=args.dist_url,
        args=(args,),
    )
```

## Expected behavior:

I would ecpect all the COCO evaluations to be pressent in the metrics.json file and on TensorBoard

## Environment:
```
Failed to load OpenCL runtime
/bin/sh: 1: /usr/local/cuda/bin/nvcc: not found
------------------------  ---------------------------------------------------------------------------------------------------------------------
sys.platform              linux
Python                    3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]
numpy                     1.13.3
detectron2                0.1.3 @/home/appuser/.local/lib/python3.6/site-packages/detectron2
detectron2 compiler       GCC 7.3
detectron2 CUDA compiler  10.2
detectron2 arch flags     /home/appuser/.local/lib/python3.6/site-packages/detectron2/_C.cpython-36m-x86_64-linux-gnu.so; cannot find cuobjdump
DETECTRON2_ENV_MODULE     <not set>
PyTorch                   1.5.1 @/home/appuser/.local/lib/python3.6/site-packages/torch
PyTorch debug build       False
CUDA available            True
GPU 0                     GeForce RTX 2080 Ti
CUDA_HOME                 /usr/local/cuda
NVCC
Pillow                    7.2.0
torchvision               0.6.1 @/home/appuser/.local/lib/python3.6/site-packages/torchvision
torchvision arch flags    /home/appuser/.local/lib/python3.6/site-packages/torchvision/_C.so; cannot find cuobjdump
fvcore                    0.1.1.post20200630
cv2                       3.2.0
------------------------  ---------------------------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,
```

**Comment by ppwwyyxx** [comment]
I think it's because they (tensorboard and metrics.json) are by default configured to dump results once every 20 iterations. We would want to address this with either a better message or a more clever user experience.

**Comment by fromm1990** [comment]
Thanks for your quick response. I was not aware of this detail. I guess if I set `EVAL_PERIOD` to something that is dividable with 20, every evaluation would be logged in metric.json and TensorBoard?

**Comment by fromm1990** [comment]
I can confirm, changing `EVAL_PERIOD` such it is dividable with 20, will log all evaluations to metric.json and TensorBoard. Maybe the CfgNode could contain an option such the log interval could be adjustable?

**Comment by ppwwyyxx** [comment]
this should be fixed by https://github.com/facebookresearch/detectron2/commit/e1252f9aab632a23c40787caf700745a2e285a58 now
